### PR TITLE
build(bazel): improve remote caching for AIO local deps build

### PR DESCRIPTION
**Pull request by kormide**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Architect tests for the AIO local deps build run on every pull request.

## What is the new behavior?

Test results are cached properly, significantly speeding up the build when there are no AIO changes.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/BUILD.bazel [hunk 1]
```diff
@@ -125,7 +125,6 @@ copy_to_directory(
 # All source and configuration files required to build the docs app
 APPLICATION_FILES = [
     "angular.json",
-    "package.json",
```
Comment (kormide) [inline aio/BUILD.bazel:128]: This wasn't a necessary input to the architect actions.

**Comment by kormide** [comment]
Note that the architect build is also not hermetic as the `ngsw.json` file it generates has a timestamp in it. I'm not sure if it's correct to remove or hardcode that timestamp, and if it's worth the effort to do that.

**Review by devversion — APPROVED** [review]
Nice, thanks!

File: aio/scripts/local-workspace-status.mjs [hunk 2]
```diff
@@ -16,7 +16,20 @@ const pkgJson = JSON.parse(fs.readFileSync(pkgJsonPath, 'utf8'));
...
+BUILD_HOST fake_host
+BUILD_USER fake_user
+`)
+
+// Fix the timestamp in volatile-status.txt to improve remote cache performance.
+// Unlike the local Bazel cache, the remote cache does not ignore volatile-status.txt
+// and will invalidate actions that depend on it when the values change.
+// https://github.com/bazelbuild/bazel/issues/10075#issuecomment-546872111
+console.log(`\
+BUILD_TIMESTAMP 0
```
Comment (devversion) [inline aio/scripts/local-workspace-status.mjs:34]: One thing I found interesting from the Bazel issue. It sounds like we shouldn't really rely on stamping for development builds. This makes a lot of sense to me. I think this is a great improvement regardless and solves the issue, but I wonder if long-term Angular should make sure AIO can be built w/ local packages unstamped? I believe we could patch/configure architect appropriately

Comment (kormide) [reply aio/scripts/local-workspace-status.mjs:34]: I agree that altering architect is probably the right long-term solution.

**Comment by devversion** [comment]
This PR was merged into the repository by commit 0ae80a78a318b7ae09c44337d21962aadea339b9.

**Pull request closed without merging**